source/cursorManager.py

```diff
@@ -284,6 +284,9 @@ def _selectionMovementScriptHelper(self,unit=None,direction=None,toPosition=None
 				# 1. Caret at 1; selection (1, 1)
 				# 2. Shift+rightArrow: selection (1, 2)
 				newInfo.setEndPoint(oldInfo, "startToStart")
+		# _isSelectionAnchoredAtStart is not the same as _lastSelectionMovedStart,
+		# i.e. their default values are opposite and _isSelectionAnchoredAtStart is only updated if the selection has changed.
+		self._isSelectionAnchoredAtStart = not self._lastSelectionMovedStart

```

Comment:
@jcsteh I think we need to make sure this is compatible with https://github.com/nvaccess/nvda/commit/62f42c611ed7383aab905b18f61293b2884ee4da

Did some testing and that looked good.